File: code_examples/3-Structured_Problem_Solving/6-If_Then/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks if there is over 30 students
 */

package main

import "fmt"

func main() {
	// this function checks if there is over 30 students
	const maxStudentNumber int = 30
	var numberOfStudents int

	// input
	fmt.Print("Enter the number of students: ")
	fmt.Scan(&numberOfStudents)

	// process
	if numberOfStudents > maxStudentNumber {
		// output
		fmt.Println("Too many students!")
	}

	fmt.Println("\nDone.")
}

File: code_examples/3-Structured_Problem_Solving/6-If_Then/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks if there is over 30 students
*/

using System;

/*
 * The Program class
*/
class Program {
    static void Main() {
        // this function checks if there is over 30 students
        const int MAX_STUDENT_NUMBER = 30;
        int numberOfStudents;

        // input
        Console.Write("Enter the number of students: ");
        numberOfStudents = Convert.ToInt32(Console.ReadLine());

        // process
        if (numberOfStudents > MAX_STUDENT_NUMBER) {
            // output
            Console.WriteLine("Too many students!");
        }

        Console.WriteLine("\nDone.");
    }
}
